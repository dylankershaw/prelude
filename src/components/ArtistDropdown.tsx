import React from 'react';

import {colors} from '../theme';

interface Props {
  selectedArtist: string;
  setSelectedArtist(artist: string): void;
  artists: Array<string>;
}

const ArtistDropdown: React.FC<Props> = ({selectedArtist, setSelectedArtist, artists}) => {
  return (
    <div className='container'>
      Select an artist:
      <select value={selectedArtist} onChange={e => setSelectedArtist(e.target.value)}>
        {artists.map((a, i) => (
          <option key={i} value={a}>
            {a}
          </option>
        ))}
      </select>
      <style jsx>
        {`
          .container {
            margin-top: 2rem;
          }

          select {
            background-color: ${colors.green};
            border-radius: 1rem;
            color: white;
            display: block;
            max-width: 100%;
          }
        `}
      </style>
    </div>
  );
};

export default ArtistDropdown;
