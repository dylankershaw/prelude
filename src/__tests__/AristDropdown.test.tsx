import React from 'react';
import {render} from '@testing-library/react';

import ArtistDropdown from '../components/ArtistDropdown';

test('matches snapshot', () => {
  const {container} = render(
    <ArtistDropdown
      selectedArtist='The Doors'
      setSelectedArtist={(a: string) => {}}
      artists={['The Who', 'The Doors', 'Lou Reed']}
    />
  );
  expect(container).toMatchSnapshot();
});
