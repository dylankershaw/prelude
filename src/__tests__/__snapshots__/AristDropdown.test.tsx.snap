// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    Select an artist:
    <select>
      <option
        value="The Who"
      >
        The Who
      </option>
      <option
        value="The Doors"
      >
        The Doors
      </option>
      <option
        value="Lou Reed"
      >
        Lou Reed
      </option>
    </select>
    <style>
      
          .container {
            margin-top: 2rem;
          }

          select {
            background-color: #1DB954;
            border-radius: 1rem;
            color: white;
            display: block;
            max-width: 100%;
          }
        
    </style>
  </div>
</div>
`;
